quipucords/scanner/network/inspect_callback.py

```diff
@@ -56,73 +58,87 @@ def v2_runner_on_ok(self, result):
             facts = {}
             for key, value in host_facts.items():
                 if key == 'host_done':
-                    facts = self._ansible_facts[host]
-                    logger.debug('host scan complete for %s with facts %s',
-                                 host, facts)
-
-                    # Update scan counts and save
-                    if self.scan_task is not None:
-                        self.scan_task.systems_scanned += 1
-                        self.scan_task.save()
-
-                    # Save facts for host
-                    sys_result = SystemInspectionResult(
-                        name=host, status=SystemInspectionResult.SUCCESS)
-                    sys_result.save()
-                    for result_key, result_value in facts.items():
-                        stored_fact = RawFact(name=result_key,
-                                              value=result_value)
-                        stored_fact.save()
-                        sys_result.facts.add(stored_fact)
-                    sys_result.save()
-
-                    inspect_result = self.inspect_results.results.filter(
-                        source__id=self.source.id).first()
-                    if inspect_result is None:
-                        inspect_result = InspectionResult(
-                            scan_task=self.scan_task, source=self.source)
-                        inspect_result.save()
-                    inspect_result.systems.add(sys_result)
-                    inspect_result.save()
-                    self.inspect_results.results.add(inspect_result)
-                    self.inspect_results.save()
-
+                    self._finalize_host(host)
```

Comment:
Is there a conceivable program flow where the host finishes before any other ansible facts get set for this host? (Thinking about a key error). Just thinking about a key error down on [line #92](https://github.com/quipucords/quipucords/pull/344/files#diff-11e150690b3e394303383728124f461fR92) that could be avoided with a self._ansible_facts.get(host)

Reply:
I don't think so.  The `host_done` is there to mark the end of of host scan.

Reply:
That seems like an edge case that could occur if the user runs a scan but specifies no facts or something like that. I will fix it, thanks!

Reply:
Done.